src/main/resources/db/migration/V1_117__update_print_files_to_clob_procedure.sql

```diff
@@ -0,0 +1,168 @@
+			VALUES (NEXTVAL('juror_mod.content_store_seq'),v_filename,'PRINT',v_data);
+			COMMIT;
+
+			-- setting new file name as last file has filled up to the limit
+			SELECT  NEXTVAL('juror_mod.print_file_count') INTO v_revision;
+			v_filename := 'JURY'||LPAD(v_revision::VARCHAR(22),4,'0')||'01.0001';
+
+			-- erasing data for new file
+			v_data := '';
+			p_count := 0;
+
+
+		END IF;
+
+		-- Form the data column details first
+		SELECT 	form_details.id,
+				RPAD('   ' ||RPAD(p_form_type,16)||LPAD(form_details.counter::VARCHAR(20),6,'0')||LPAD(form_details.counter::VARCHAR(20),6,'0')||'50'||LPAD(p_max_rec_len::VARCHAR(20),8,'0'),256,' '),
+				form_details.detail_rec
+		INTO v_row_id, v_header, v_detail_rec;
+
+		SELECT LENGTH(form_details.detail_rec)
+		INTO v_max_data_len;
+
+		-- round up the division to the nearest whole number - e.g. 1.4 would equate to 2
+		SELECT CEIL(v_max_data_len/p_max_rec_len)
+		INTO v_max_loop_count;
+
+		-- If data is empty then the first line will be the header plus newline otherwise it will be the current data plus header plus newline
+		v_data := v_data||v_header||CHR(10);
+
+		-- reset the starting string position
+		v_strpos := 1;
+		v_count := 1;
+		-- Multiple lines for the details under the header if it exceeds the max record threshold
+		FOR v_count IN 1..v_max_loop_count loop
+			IF LENGTH(v_detail_rec) > 0  THEN
+				v_data := v_data||v_detail_rec||CHR(10); -- add line details to the data column
+			END IF;
+			v_strpos := v_strpos + p_max_rec_len;  -- increment the string position identifier so that the next line starts from the end of the previous line counter
+	 	END LOOP;
+
+
+		UPDATE juror_mod.bulk_print_data
+		SET  extracted_flag = 'Y'
+		WHERE id = v_row_id;
+
+		-- increment the loop counter so that the process can be stopped if it reaches the threshold
+		p_count := p_count + 1;
+
+	END LOOP;
+END;
+$procedure$
+;
+
+-- DROP PROCEDURE juror_mod.delete_printfiles();
+
+CREATE OR REPLACE PROCEDURE juror_mod.delete_printfiles()
+ LANGUAGE plpgsql
+AS $procedure$
+BEGIN
+	DELETE
+	FROM juror_mod.bulk_print_data bpd
+	WHERE COALESCE (bpd.extracted_flag,'N') = 'N'
+	AND EXISTS 	(
+					SELECT 1
+					FROM juror_mod.juror_pool jp
+					WHERE jp.juror_number = bpd.juror_no
+					AND jp.is_active = true
+					AND (
+							(bpd.form_type IN ('5224','5224C') AND jp.status <> 6) -- withdrawal letters
+			                OR
+			                (bpd.form_type IN ('5225','5225C') AND jp.status <> 5) -- excusal letters
+			                OR
+			                (bpd.form_type IN ('5226','5226C') AND jp.status <> 2) -- excusal denied letters
+			                OR
+			                (bpd.form_type IN ('5226A','5226AC') AND jp.status <> 2) -- deferral denied letters
+			                OR
+			                (bpd.form_type IN ('5227','5227C') AND jp.status <> 1) -- request for info letters
+			                OR
+			                (bpd.form_type IN ('5229','5229C') AND jp.status <> 7) -- postpone letters
```

Comment:
@ChrisWCGI - how come postpone letters don't check for responded status like deferral letters do?

Reply:
This is Heritage not being consistent. We have chosen not to change/fix it here.